pkg/config/engine/api.go

```diff
@@ -23,4 +23,5 @@ type Interface interface {
 	Set(string, interface{})
 	RemoveRuntime(string) error
 	Save(string) (int64, error)
+	GetRuntime(name string) (interface{}, error)
```

Comment:
This should be `GetRuntimeConfig`.

Returning an `interface{}` type here is not ideal. At the very least we should return a `map[string]interface{}` here instead to indicate that we're returning a nested key-value config.

An alternative is to add a `Runtime` interface type that we can add the functionality to that we required (get binaries / paths in this case).

Reply:
What are your thoughts on adding a generic `RuntimeData` struct which will act as an adapter object? The `RuntimeData` struct could encapsulate all the common and relevant fields across the different container runtimes

Some of the common fields in both cri-o and containerd that i've observed:

- Binary Path
- Runtime Type
- Runtime Root